### DISCUSSION TITLE
Is it possible to get a fuzzed-data database based on the default value?

### DISCUSSION THREAD

**Discussion by User-debbie** [jtpereyda/boofuzz#575] [Q&A]
Is it possible to get a **database of fuzzed-data generated** from boofuzz?

For example I have an web API to fuzz, and I only **need to fuzz JSON body** since the body is submmited by the users, and I only need to fuzz the filed called `AnswerContent`.

Here **I have a complete procedure of the API**, including login,choose form and etc. which just need me to pass in 'AnswerContent' as variable. 

In this case, I'd like to use every test case generated by boofuzz as input.

Let's say `this is an answer` is the default value of variable `AnswerContent`

In other fuzzer, e,g, SecRayFuzz, Peach Fuzzer，a huge amount of test cases would be generated by various mutators, for example `Duplicate\Remove\Swap...`. In that case I can get test cases like
`this is an a` or  `tttt is an answer` or `@#$$ is an &*(`, somthing like that

Then I can use these tests cases as input, and let the procedure finish the rest.

In a word, I just want to get a fuzzed-data base, which mutates from the default value I've provided. In this way, boofuzz would become a strong plug-in of my system, and I can cooperate more efficiently with the current system.

Thx for helping!

**Comment by cq674350529** [comment]
If you just want to get all the test cases generated by `boofuzz`, not use the `boofuzz` framework itself, you can copy and re-use the code from `s_string()` to achieve this. 

By the way, maybe you can use [radamsa](https://gitlab.com/akihe/radamsa) to generate mutated data purely, which is a test case generator for robustness testing.

**Comment by User-debbie** [comment]
@cq674350529 Thanks! Using the s_string() code seems very efficient! 
And about yesterday's question about REAL RANDOM I mentioned in [#477](https://github.com/jtpereyda/boofuzz/issues/477) 
I just felt confused because generated test case are all like `%01%02%03@%04%0a%0d%0aADSF` or `&reboot`, which is not relevant to the `default value` I gave. Now I know after reading the definition of primitive string, I found `_fuzz_library` in it! Does that mean if I need the test cases to mutate base on the `default value`, I should add this `default value` to the _fuzz_library?

**Comment by User-debbie** [comment]
Well, I tried to delete things in `long_string_seeds` and `_fuzz_library` in `string.py`, leaving `change of fuzz library`  alone in the  `_fuzz_library`.

value `boofuzz` is the default value of the fuzzed field

As a result, I get fuzzed data like `DD\\x00DDDDD`,`boofuzzboofuzzboofuzzboofuzzboofuzzboofuzzboofuzzb`, 23 in total, sadly not exactly what I wanted.

I thought I could get something like `change of fuzz library change of fuzz library change of fuzz library`  or `chhhhange of fuzz li￥￥%……ry`

**Comment by cq674350529** [comment]
@User-debbie Some of test cases are related to the `default value`, some are related to the `_fuzz_library` and length variation. The mutate strategy of `s_string()` is able to trigger bugs such as `buffer overflow`, `command injection` in common cases. As to the `buffer overflow`, I believe there is no difference between `change of fuzz library change of fuzz library change of fuzz library` and `'a' * 0x100`.

Based on your goal, you can add custom fuzz values using the parameter `fuzz_values` of `s_string()` (I didn't try this way before). To generate data like `chhhhange of fuzz li￥￥%……ry`, maybe you can try `radamsa`, then feed it into `boofuzz`.

**Comment by User-debbie** [comment]
@cq674350529 Thanks! Good idea,maybe I can use `radamsa` to generate value and add them to `_fuzz_library`.?

**Comment by jtpereyda** [comment]
@User-debbie 

There is also [FileConnection](https://github.com/jtpereyda/boofuzz/blob/master/boofuzz/connections/file_connection.py) which will write each fuzz case to a file within a directory. That's a simple way to get a dump of all the mutation outputs. However if using the string primitive code works, that's solid too.

>value boofuzz is the default value of the fuzzed field
>
> As a result, I get fuzzed data like DD\\x00DDDDD,boofuzzboofuzzboofuzzboofuzzboofuzzboofuzzboofuzzb, 23 in total, sadly not exactly what I wanted.

To get what you're going for, I would pass in "change of fuzz library" as the default value.
